public/app/features/explore/state/reducers.ts

```diff
@@ -114,6 +116,7 @@ export const makeExploreItemState = (): ExploreItemState => ({
   mode: null,
   isLive: false,
   urlReplaced: false,
+  queryState: new PanelQueryState(),
```

Comment:
Not sure about this. I feel like this should not be accessible by the react and also does not drive rendering in any. We do not change it and it is also mutable. So I wonder if we shouldn't just move it outside of the redux store and handle it as external dependency kind of thing.

Reply:
Think this is fine for now, it is mutable so not ideal to have in redux but not forbidden either. Think before we do more complex work figure out the proper redux solution around this we can move ahead with this. I think the coming rewrite to how queries are executed to use observable patterns, that will fundamentally change things so good to wait to do more work around the current API & responsibility of PanelQueryState as I think that will change

Reply:
Good points! But I'll let it be for now.